lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -140,3 +148,66 @@ defaultRetryPolicy =
     limitRetriesByCumulativeDelay (60 * second) (exponentialBackoff 10000)
   where
     second = 1000*1000
+
+type RestoreHandle block = (NE.NonEmpty block -> BlockHeader -> IO Bool)
+
+class Cursor s where
+    toHeader :: s -> BlockHeader
+
+restore
```

Comment:
What do you think of changing the name here? Two reasons:

- Restore is so-to-speak a wallet concept, doesn't really fit in the network layer
- It's not really about restoration anymore since, the actual restoration part has been parameterized away

What about something like "follow" or "chainSync" ?